ion-schema/src/isl/isl_constraint.rs

```diff
@@ -645,6 +660,42 @@ impl IslConstraintImpl {
                     }
                 }
             }
+            "field_names" => {
+                let type_reference =
+                    IslTypeRefImpl::from_ion_element(isl_version, value, inline_imported_types)?;
+                match isl_version {
+                    IslVersion::V1_0 => {
+                        // for ISL 1.0 `field_names` constraint does not exist hence `field_names` will be considered as open content
+                        Ok(IslConstraintImpl::Unknown(
+                            constraint_name.to_string(),
+                            value.to_owned(),
+                        ))
+                    }
+                    IslVersion::V2_0 => {
+                        // return error if there are any annotations other than `distinct`
+                        if value.annotations().any(|a| a.text() != Some("distinct")) {
+                            return Err(invalid_schema_error_raw(
+                                "field_names constraint can only contain `distinct` annotation",
+                            ));
+                        }
```

Comment:
Suggestion—make this code like the code for `fields`. It ensures that there's at most 1 occurrence of "distinct".
```suggestion
                        if value.annotations().count() > 1
                            || value.annotations().any(|a| a.text() != Some("distinct"))
                        {
                            return invalid_schema_error(
                                "field_names constraint may only be annotated with 'distinct'",
                            );
                        }
```